### ISSUE TITLE
Automatically build app on push

### ISSUE THREAD

**Issue by itsalaidbacklife** [Human-AI-Interaction/cuttle-bot#7]
cuttle-bot is hosted on github pages and the app is served out of the `docs` folder. Currently, deploying the app requires manually running the build command, copying the `dist` output into `docs`, committing the change, and pushing to `master`.  We should automate this process so that merges into `master` are automatically deployed (similar to the main [cuttle repo](https://github.com/cuttle-cards/cuttle)). To do so, we should:

1. Update `angular.json` so the build `outpath` is `docs` instead of it's current `dist` (this way builds are put in the correct folder)
2. Add a new `.github/workflows/build.yml` action file that runs whenever a commit is made to any branch except`master` and runs the build command, commits the changes, and pushes them to the current branch. This way when we merge into `master`, the commit will include the build of the latest source and this will automatically update the deployment

**Comment by JonasThers** [comment]
I "blindly" took a shot at this, haven't tried anything like this before

**Comment by itsalaidbacklife** [comment]
@JonasThers I'm reopening because whatever the "pages build and deployment" action is doing it isn't working, so there is still stuff to figure out about this

**Comment by JonasThers** [comment]
Hmmm, any other idea how to get around this?

**Comment by itsalaidbacklife** [comment]
@JonasThers yes! I have generally done less maintenance on this repo than the PVP one so I'm embarrassed to admit I literally forgot how to build and deploy it... but I've figured it out! The background update is live and now we can look at automating the deployment process. I think we basically need to run the build command when a PR is merged and commit + push the changes

**Comment by itsalaidbacklife** [comment]
It looks like github pages has a new approach to deployment that uses github actions instead of statically deploying the 'docs' folder of a given branch, so that might be a good way to set this up. In the mean time I've got a build process set up again so if you'd like to tackle #3 or anything else in this repo I'm able to deploy the changes now 😅
